src/NServiceBus.NHibernate/SynchronizedStorage/NHibernateAmbientTransactionSynchronizedStorageSession.cs

```diff
@@ -23,14 +23,20 @@ public NHibernateLazyAmbientTransactionSynchronizedStorageSession(Func<IDbConnec
 
         public void Dispose()
         {
-            if (session.IsValueCreated)
+            try
```

Comment:
shouldn't a single IsValueCreated be outside the tryf? rather then two checks inside?